packages/jaeger-ui/src/components/TracePage/CriticalPath/index.tsx

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2023 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import memoizeOne from 'memoize-one';
+import { Span, Trace, criticalPathSection } from '../../../types/trace';
+import removeFollowFromChildSpans from './utils/getChildOfSpans';
+import findLastFinishingChildSpanId from './utils/findLastFinishingChildSpanId';
+import findRootSpanId from './utils/findRootSpanId';
+import sanitizeOverFlowingChildren from './utils/sanitizeOverFlowingChildren';
+
+export const computeCriticalPath = (
+  traceData: Trace,
+  spanId: string,
+  criticalPath: criticalPathSection[],
+  spawnTime?: number
+) => {
+  // spawnTime refers to spawn synchronization event of current span
```

Comment:
the definition is circular, I don't follow what it means

Reply:
`Role of spawnTime`: I explained in this comment ,Please take a look and let me know if you need any further clarifications.
https://github.com/jaegertracing/jaeger-ui/pull/1532#discussion_r1285084429

Reply:
I agree with @yashrsharma44 - he asked to add that explanation as comments in the code, not a link to a discussion on GH.

Having said that, I still don't understand `spawnTime` - can you provide a clear definition of what that is? For example, as a user of this `computeCriticalPath` function, what am I supposed to pass for `spawnTime`? In fact, the whole function signature could use help explaining what each parameter means.

Reply:
https://github.com/jaegertracing/jaeger-ui/blob/97d8eefe243fcdba1250dabd5bbb60b9cb410c68/packages/jaeger-ui/src/components/TracePage/CriticalPath/index.tsx#L21-L41
Is this function definition understandable??